src/core/thread/mle.cpp

```diff
@@ -1401,17 +1405,13 @@ ThreadError Mle::SendParentRequest(void)
         mParentRequest.mChallenge[i] = static_cast<uint8_t>(otPlatRandomGet());
     }
 
-    VerifyOrExit((message = NewMleMessage()) != NULL, ;);
-    SuccessOrExit(error = AppendHeader(*message, Header::kCommandParentRequest));
-    SuccessOrExit(error = AppendMode(*message, mDeviceMode));
-    SuccessOrExit(error = AppendChallenge(*message, mParentRequest.mChallenge, sizeof(mParentRequest.mChallenge)));
-
     switch (mParentRequestState)
     {
     case kParentRequestRouter:
         scanMask = ScanMaskTlv::kRouterFlag;
 
-        if (mParentRequestMode == kMleAttachSamePartition)
+        if (mParentRequestMode == kMleAttachSamePartition1 ||
+            mParentRequestMode == kMleAttachSamePartition2)
```

Comment:
Attaching to a better partition only happens in response to receiving an MLE Advertisement that indicates a better partition.  In this case, I think we should first search for Routers and then Routers and REEDs.